cache/cache_internal_test.go

```diff
@@ -0,0 +1,569 @@
+	require.Equal(t, len(testData), testSize)
+	require.NoError(t, err)
+
+	// write the object
+	o := writeObjectBytes(t, cfs.UnWrap(), "data.bin", testData)
+	require.Equal(t, o.Size(), int64(testSize))
+
+	o2, err := rootFs.NewObject("data.bin")
+	require.NoError(t, err)
+	require.Equal(t, o2.Size(), o.Size())
+
+	// check sample of data from in-file
+	sampleStart := chunkSize / 2
+	sampleEnd := chunkSize
+	testSample := testData[sampleStart:sampleEnd]
+	checkSample := readDataFromObj(t, o2, sampleStart, sampleEnd, false)
+	require.Equal(t, len(checkSample), len(testSample))
+
+	for i := 0; i < len(checkSample); i++ {
+		require.Equal(t, testSample[i], checkSample[i])
+	}
+}
+
+func TestInternalWrappedFsChangeNotSeen(t *testing.T) {
+	reset(t)
+	cfs, err := getCacheFs(rootFs)
+	require.NoError(t, err)
+	chunkSize := cfs.ChunkSize()
+
+	// create some rand test data
+	co := writeObjectRandomBytes(t, rootFs, (chunkSize*4 + chunkSize/2))
+
+	// update in the wrapped fs
+	o, err := cfs.UnWrap().NewObject(co.Remote())
+	require.NoError(t, err)
+	err = o.SetModTime(co.ModTime().Truncate(time.Hour))
+	require.NoError(t, err)
+
+	// get a new instance from the cache
+	co2, err := rootFs.NewObject(o.Remote())
+	require.NoError(t, err)
+
+	require.NotEqual(t, o.ModTime(), co.ModTime())
+	require.NotEqual(t, o.ModTime(), co2.ModTime())
+	require.Equal(t, co.ModTime(), co2.ModTime())
+}
+
+func TestInternalChangeSeenAfterDirCacheFlush(t *testing.T) {
+	cfs, err := getCacheFs(rootFs)
+	require.NoError(t, err)
+
+	cfs.DirCacheFlush() // flush the cache
+
+	l, err := cfs.UnWrap().List("")
+	require.NoError(t, err)
+	require.Len(t, l, 1)
+	o := l[0]
+
+	// get a new instance from the cache
+	co, err := rootFs.NewObject(o.Remote())
+	require.NoError(t, err)
+	require.Equal(t, o.ModTime(), co.ModTime())
+}
+
+func TestInternalWarmUp(t *testing.T) {
+	reset(t)
+	cfs, err := getCacheFs(rootFs)
+	require.NoError(t, err)
+	chunkSize := cfs.ChunkSize()
+
+	o1 := writeObjectRandomBytes(t, rootFs, (chunkSize * 3))
+	o2 := writeObjectRandomBytes(t, rootFs, (chunkSize * 4))
+	o3 := writeObjectRandomBytes(t, rootFs, (chunkSize * 6))
+
+	_ = readDataFromObj(t, o1, 0, chunkSize, false)
+	_ = readDataFromObj(t, o2, 0, chunkSize, false)
+
+	// validate a fresh chunk
+	expectedExpiry := time.Now().Add(chunkAge)
+	ts, err := boltDb.GetChunkTs(path.Join(rootFs.Root(), o2.Remote()), 0)
+	require.NoError(t, err)
+	require.WithinDuration(t, expectedExpiry, ts, time.Second)
+
+	// validate that we entered a warm up state
+	_ = readDataFromObj(t, o3, 0, chunkSize, false)
+	require.True(t, cfs.InWarmUp())
+	expectedExpiry = time.Now().Add(metaAge)
+	ts, err = boltDb.GetChunkTs(path.Join(rootFs.Root(), o3.Remote()), 0)
+	require.NoError(t, err)
+	require.WithinDuration(t, expectedExpiry, ts, time.Second)
+
+	// validate that we cooled down and exit warm up
+	// we wait for the cache to expire
+	fmt.Printf("Waiting 10 seconds for warm up to expire\n")
```

Comment:
This should be t.Logf() rather than fmt.Printf so it is quiet if tests are run without -v

Reply:
Done